### PR TITLE
CDRIVER-5721 document bson_validate_with_error_and_offset()

### PR DESCRIPTION
**Pull request by mdbmes**
Adds the missing documentation page for `bson_validate_with_error_and_offset()`. This also moves `bson_validate_flags_t` to its own page, and clarifies the different meanings of `0` here in libbson vs libmongoc.

This patch re-runs the `build/generate-opts` script, which also caused minor unrelated diffs in `mongoc-opts.c`.

### CODE REVIEW

File: src/libmongoc/src/mongoc/mongoc-opts-private.h
Comment (rcsanchez97) [inline src/libmongoc/src/mongoc/mongoc-opts-private.h:1]: The license header needs to be restored.

**Review by rcsanchez97 — APPROVED** [review]
LGTM, modulo the one license header that needs to be restored.

File: src/libmongoc/src/mongoc/mongoc-opts-private.h
Comment (mdbmes) [reply src/libmongoc/src/mongoc/mongoc-opts-private.h:1]: This file is automatically generated, I just re-ran the generator. It looks like someone added this header without including it on the template as well. I'll fix that. Thanks.

**Comment by mdbmes** [comment]
Thanks for the review, Roberto!

**Pull request merged**